sys/kern/interrupt.c

```diff
@@ -127,60 +141,103 @@ void intr_root_handler(ctx_t *ctx) {
     on_user_exc_leave();
 }
 
-/* interrupt handlers delegated to be called in the interrupt thread */
-static ih_list_t delegated = TAILQ_HEAD_INITIALIZER(delegated);
+static void intr_thread_create(intr_event_t *ie) {
+  intr_thread_t *it = kmalloc(M_INTR, sizeof(intr_thread_t), M_ZERO);
+
+  assert(it);
+
+  ie->ie_ithread = it;
```

Comment:
You're missing `it->it_event = ie`. For now every interrupt thread has associated interrupt event set to NULL.